File: Backend/AssignmentManagementSystem.BusinessLogicLayer/Entities/Submission.cs
using System;
using System.Collections.Generic;

namespace AssignmentManagementSystem.BusinessLogicLayer.Entities;

public partial class Submission
{
    public Guid Id { get; set; }
    public Guid AssignmentId { get; set; }
    public Guid StudentId { get; set; }
    public string? SubmissionText { get; set; }
    public string? SubmissionFileUrl { get; set; }
    public DateTime SubmittedAt { get; set; }
    public string? Status { get; set; }
    public decimal? MarksObtained { get; set; }
    public string? FeedBack { get; set; }
    public DateTime? GradedAt { get; set; }
    public Guid? GradedBy { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public bool IsActive { get; set; }

    public virtual Assignment Assignment { get; set; } = null!;
    public virtual User? GradedbyNavigation { get; set; }
    public virtual User Student { get; set; } = null!;
}

File: Frontend/src/types/submission.ts
export interface Submission {
    id: string;
    assignmentId: string;
    assignmentTitle: string;
    studentId: string;
    studentName: string;
    studentEmail: string;
    submissionText?: string;
    submissionFileUrl?: string;
    submittedAt: string;
    status: string;
    marksObtained?: number;
    feedback?: string;
    gradedAt?: string;
    gradedBy?: string;
    isActive: boolean;
    createdAt: string;
    updatedAt?: string;
}

export interface SubmissionGradeData {
    submissionId: string;
    marksObtained: number;
    feedback?: string;
    gradedBy: string;
    status: string;
}

export interface SubmissionGroup {
    assignmentId: string;
    assignmentTitle: string;
    teacherId: string;
    teacherName: string;
    classId: string;
    className: string;
    subjectId: string;
    subjectName: string;
    dueDate?: string;
    totalMarks?: number;
    submissions: Submission[];
}

export interface CreateSubmissionData {
    assignmentId: string;
    studentId: string;
    submissionText?: string;
    submissionFileUrl?: string;
}

export interface GradeResponse {
    success: boolean;
    message: string;
    data: Submission;
}